### ISSUE TITLE
App Crashes on selecting time API 14

### ISSUE THREAD

**Issue by nyayadhish20** [wdullaer/MaterialDateTimePicker#307]
Full Stacktrace:- 
02-01 13:34:49.567 12277-12277/com.heallifydoctor W/dalvikvm: VFY: unable to resolve static field 10470 (ACTION_SCROLL_BACKWARD) in Landroid/view/accessibility/AccessibilityNodeInfo$AccessibilityAction;
02-01 13:34:49.567 12277-12277/com.heallifydoctor I/dalvikvm: Could not find method android.widget.FrameLayout.performAccessibilityAction, referenced from method com.wdullaer.materialdatetimepicker.time.RadialPickerLayout.performAccessibilityAction
02-01 13:34:49.567 12277-12277/com.heallifydoctor W/dalvikvm: VFY: unable to resolve virtual method 22637: Landroid/widget/FrameLayout;.performAccessibilityAction (ILandroid/os/Bundle;)Z
02-01 13:34:49.617 12277-12277/com.heallifydoctor I/dalvikvm: Could not find method android.view.View.announceForAccessibility, referenced from method com.wdullaer.materialdatetimepicker.Utils.tryAccessibilityAnnounce
02-01 13:34:49.617 12277-12277/com.heallifydoctor W/dalvikvm: VFY: unable to resolve virtual method 21630: Landroid/view/View;.announceForAccessibility (Ljava/lang/CharSequence;)V
02-01 13:34:55.887 12277-12277/com.heallifydoctor E/RadialTextsView: RadialTextView was not ready for animation.
02-01 13:34:55.887 12277-12277/com.heallifydoctor E/RadialSelectorView: RadialSelectorView was not ready for animation.
02-01 13:34:55.917 12277-12277/com.heallifydoctor W/dalvikvm: threadid=1: thread exiting with uncaught exception (group=0x409961f8)
02-01 13:34:55.987 12277-12277/com.heallifydoctor E/AndroidRuntime: FATAL EXCEPTION: main
                                                                    java.lang.IllegalStateException: Circular dependencies cannot exist in AnimatorSet
                                                                        at android.animation.AnimatorSet.sortNodes(AnimatorSet.java:816)
                                                                        at android.animation.AnimatorSet.start(AnimatorSet.java:454)
                                                                        at com.wdullaer.materialdatetimepicker.time.RadialPickerLayout.setCurrentItemShowing(RadialPickerLayout.java:682)
                                                                        at com.wdullaer.materialdatetimepicker.time.TimePickerDialog.setCurrentItemShowing(TimePickerDialog.java:1189)
                                                                        at com.wdullaer.materialdatetimepicker.time.TimePickerDialog.advancePicker(TimePickerDialog.java:1010)
                                                                        at com.wdullaer.materialdatetimepicker.time.RadialPickerLayout.onTouch(RadialPickerLayout.java:857)
                                                                        at android.view.View.dispatchTouchEvent(View.java:5481)
                                                                        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:1953)
                                                                        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1714)
                                                                        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:1959)
                                                                        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1728)
                                                                        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:1959)
                                                                        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1728)
                                                                        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:1959)
                                                                        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1728)
                                                                        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:1959)
                                                                        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1728)
                                                                        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:1959)
                                                                        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1728)
                                                                        at com.android.internal.policy.impl.PhoneWindow$DecorView.superDispatchTouchEvent(PhoneWindow.java:1892)
                                                                        at com.android.internal.policy.impl.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1371)
                                                                        at android.app.Dialog.dispatchTouchEvent(Dialog.java:725)
                                                                        at com.android.internal.policy.impl.PhoneWindow$DecorView.dispatchTouchEvent(PhoneWindow.java:1840)
                                                                        at android.view.View.dispatchPointerEvent(View.java:5662)
                                                                        at android.view.ViewRootImpl.deliverPointerEvent(ViewRootImpl.java:2863)
                                                                        at android.view.ViewRootImpl.handleMessage(ViewRootImpl.java:2442)
                                                                        at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                        at android.os.Looper.loop(Looper.java:137)
                                                                        at android.app.ActivityThread.main(ActivityThread.java:4340)
                                                                        at java.lang.reflect.Method.invokeNative(Native Method)
                                                                        at java.lang.reflect.Method.invoke(Method.java:511)
                                                                        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:784)
                                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:551)
                                                                        at dalvik.system.NativeStart.main(Native Method)

**Comment by nyayadhish20** [comment]
Thank you very much....

**Comment by nyayadhish20** [comment]
Its still crashing on API 14 (4.0.1) with same errors. Do I have to change other implementation?

**Comment by wdullaer** [comment]
I haven't released this to maven central yet.
It would be great if you could try with the current master branch.

It should no longer crash, but I'm not entirely sure the behaviour will be sane on API 14 and 15 (I no longer have any devices with this API level to test with)

**Comment by nyayadhish20** [comment]
I tried to update library in my android project but didn't see the committed changes in it.
I tested with on devices of API 14 its still crash when I select the time.

**Comment by wdullaer** [comment]
If you're not seeing the committed changes, it is to be expected that it will still crash.

I will release a new version today or tomorrow, so you can pull the updated version from jcenter.

**Comment by nyayadhish20** [comment]
thank you very much...
I will check it on tomorrow.
Will let you know...

**Comment by NCrouther** [comment]
I am seeing a crash with 3.1.2 when running on an Android 4.0 emulator:

`02-20 12:12:54.674 1584-1584/com.freestylelibre.app.gb.debug E/RadialTextsView: RadialTextView was not ready for animation.
02-20 12:12:54.674 1584-1584/com.freestylelibre.app.gb.debug E/RadialSelectorView: RadialSelectorView was not ready for animation.
02-20 12:12:54.684 1584-1584/com.freestylelibre.app.gb.debug D/AndroidRuntime: Shutting down VM
02-20 12:12:54.694 1584-1584/com.freestylelibre.app.gb.debug W/dalvikvm: threadid=1: thread exiting with uncaught exception (group=0x409961f8)
02-20 12:12:54.724 1584-1584/com.freestylelibre.app.gb.debug E/AndroidRuntime: FATAL EXCEPTION: main
                                                                               java.lang.IllegalStateException: Circular dependencies cannot exist in AnimatorSet
                                                                                   at android.animation.AnimatorSet.sortNodes(AnimatorSet.java:816)
                                                                                   at android.animation.AnimatorSet.start(AnimatorSet.java:454)
                                                                                   at com.wdullaer.materialdatetimepicker.time.RadialPickerLayout.setCurrentItemShowing(RadialPickerLayout.java:682)
                                                                                   at com.wdullaer.materialdatetimepicker.time.TimePickerDialog.setCurrentItemShowing(TimePickerDialog.java:1191)
                                                                                   at com.wdullaer.materialdatetimepicker.time.TimePickerDialog.advancePicker(TimePickerDialog.java:1010)
                                                                                   at com.wdullaer.materialdatetimepicker.time.RadialPickerLayout.onTouch(RadialPickerLayout.java:873)
                                                                                   at android.view.View.dispatchTouchEvent(View.java:5481)
                                                                                   at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:1953)
                                                                                   at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1714)
                                                                                   at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:1959)
                                                                                   at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1728)
                                                                                   at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:1959)
                                                                                   at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1728)
                                                                                   at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:1959)
                                                                                   at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1728)
                                                                                   at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:1959)
                                                                                   at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1728)
                                                                                   at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:1959)
                                                                                   at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1728)
                                                                                   at com.android.internal.policy.impl.PhoneWindow$DecorView.superDispatchTouchEvent(PhoneWindow.java:1892)
                                                                                   at com.android.internal.policy.impl.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1371)
                                                                                   at android.app.Dialog.dispatchTouchEvent(Dialog.java:725)
                                                                                   at com.android.internal.policy.impl.PhoneWindow$DecorView.dispatchTouchEvent(PhoneWindow.java:1840)
                                                                                   at android.view.View.dispatchPointerEvent(View.java:5662)
                                                                                   at android.view.ViewRootImpl.deliverPointerEvent(ViewRootImpl.java:2863)
                                                                                   at android.view.ViewRootImpl.handleMessage(ViewRootImpl.java:2442)
                                                                                   at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                   at android.os.Looper.loop(Looper.java:137)
                                                                                   at android.app.ActivityThread.main(ActivityThread.java:4340)
                                                                                   at java.lang.reflect.Method.invokeNative(Native Method)
                                                                                   at java.lang.reflect.Method.invoke(Method.java:511)
                                                                                   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:784)
                                                                                   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:551)
                                                                                   at dalvik.system.NativeStart.main(Native Method)

I can launch the TimePickerDialog, but when I change the hour by tapping one of the numbers in the circle, I get the above crash.`

**Comment by NCrouther** [comment]
I think this is similar to https://github.com/code-troopers/android-betterpickers/issues/96

When I step through the code in RadialPickerLayout.java, anims[2] and anims[3] are null when running on the 4.0 emulator (which crashes) and are nonnull when running on a 5.1 emulator (which doesn't crash).
